sale_variant_configurator/models/sale_order.py

```diff
@@ -114,8 +114,34 @@ def _onchange_product_tmpl_id_configurator(self):
 
     @api.onchange('product_uom', 'product_uom_qty')
     def product_uom_change(self):
-        # TODO: Handle this when there's no product_id filled
-        super(SaleOrderLine, self).product_uom_change()
+        if self.product_id:
+            return super(SaleOrderLine, self).product_uom_change()
+        if self.product_tmpl_id and (self.order_id.pricelist_id and
```

Comment:
Incorrect indentation. When you have a multi-line `if`, always enclose full expression in parenthesis:
```
if (self.product_tmpl_id and self.order_id.pricelist_id and
        self.order_id.partner_id):
    ...
```